src/components/markdown_editor/markdown_editor.tsx

```diff
@@ -378,7 +378,10 @@ export const EuiMarkdownEditor = forwardRef<
 
     useEffect(() => {
       if (isPreviewing && autoExpandPreview && height !== 'full') {
-        if (previewRef.current!.scrollHeight > currentHeight) {
+        if (
+          currentHeight !== 'full' &&
+          previewRef.current!.scrollHeight > currentHeight
+        ) {
```

Comment:
[very very optional/nitty]

I almost think I'd rather put the `currentHeight !== 'full'` check next to the `height !== 'full'` check for symmetry - WDYT?

```suggestion
      if (
        isPreviewing &&
        autoExpandPreview &&
        height !== 'full' &&
        currentHeight !== 'full'
      ) {
        if (previewRef.current!.scrollHeight > currentHeight) {
```

Reply:
Oh, or actually early returns might make even more sense

```suggestion
      if (height === 'full' || currentHeight === 'full') return;
      if (
        isPreviewing &&
        autoExpandPreview &&
        previewRef.current!.scrollHeight > currentHeight
      ) {
```

with the above, we're down to just 1 conditional where all the conditions are more specific to the logic being performed 🤔

Reply:
Fixed in https://github.com/elastic/eui/pull/7340/commits/621a97e47eed8e922d9af5fdb9b62915e2bb568a